components/style/selector_impl.rs

```diff
@@ -82,10 +101,43 @@ impl SelectorImpl for ServoSelectorImpl {
     fn parse_pseudo_element(_context: &ParserContext,
                             name: &str) -> Result<PseudoElement, ()> {
         use self::PseudoElement::*;
-        match_ignore_ascii_case! { name,
-            "before" => Ok(Before),
-            "after" => Ok(After),
-            _ => Err(())
-        }
+        let pseudo_element = match_ignore_ascii_case! { name,
+            "before" => Before,
+            "after" => After,
+            "first-line" => FirstLine,
```

Comment:
Sure, sorry! Just parsed it in order to ensure that the problem with the first approaches was solved.